pulp_ansible/app/galaxy/v1/tasks.py

```diff
@@ -0,0 +1,229 @@
+        # parse out the role metadata
+        role_meta = get_path_role_meta(checkout_path)
+        role_tags = role_meta.get("galaxy_info", {}).get("galaxy_tags", [])
+        if role_tags is None:
+            role_tags = []
+        """
+
+        # pulp_ansible does not have a model or a view for readme content
+        """
+        ldr = RoleLoader(
+            content_type="role",
+            root=os.path.dirname(checkout_path),
+            rel_path=os.path.basename(checkout_path),
+        )
+        readme = ldr._get_readme()
+        readme_html = markup_utils._render_from_markdown(readme)
+        """
+
+        metadata = {
+            "github_user": github_user,
+            "github_repo": github_repo,
+            "role_name": role_name,
+            "name": role_name,
+            "namespace": github_user,
+            "role_version": github_reference,
+        }
+
+        # This is a pulp'ism for the source url
+        # but we will generate the download_url in the
+        # serializer so this wouldn't ever be
+        # used ... ?
+        relative_path = "%s/%s/%s.tar.gz" % (
+            metadata["github_user"],
+            metadata["role_name"],
+            metadata["role_version"],
+        )
+
+        # Make a tarball
+        shutil.rmtree(os.path.join(checkout_path, ".git"))
+        tarfn = f"/tmp/{github_user}.{role_name}.{github_reference}.tar.gz"
+        cmd = f"tar czvf {tarfn} {checkout_path}"
+        pid = subprocess.run(cmd, shell=True)
+        assert pid.returncode == 0
+
+        """
+        # pulp_ansible's Roles are actuallly Role Versions
+        # The roles list viewset aggregates the versions together
+        # by using a {namespace}.{name} as the primary key.
+        role = Role(
+            version=metadata["role_version"],
+            name=metadata["name"],
+            namespace=metadata["namespace"],
+        )
+        role.save()
+
+        # Create a real artifact
+        artifact = Artifact.init_and_validate(tarfn)
+        artifact.save()
+
+        # Use Content to tie the role and artifact together
+        ca1 = ContentArtifact.objects.create(
+            artifact=artifact, content=role, relative_path=relative_path
+        )
+        ca1.save()
+
+        # Add the role to the legacy repository via a new version.
+        legacy = AnsibleRepository.objects.get(pulp_id=ansible_repo_id)
+        add_and_remove(legacy.pk, [role.pk], [])
+        """
+
+        # Create a real artifact
+        artifact = Artifact.init_and_validate(tarfn)
+        artifact.save()
+
+        # Create the role and bind with the artifact as a "content artifact"
+        role, content_artifact = _create_role_and_content_artifact(
+            metadata, relative_path, artifact
+        )
+
+        # Add the role to the legacy repository via a new version.
+        legacy = AnsibleRepository.objects.get(pulp_id=ansible_repo_id)
+        add_and_remove(legacy.pulp_id, [role.pulp_id], [])
```

Comment:
My previous comment was meant to show that this query is unnecessary. You can just pass the `ansible_repo_id` to the `add_and_remove` call.